README.md

```diff
@@ -1,78 +1,263 @@
-$ sudo sysctl -w fs.inotify.max_user_watches=52428
+#### *file:getUploadUrl*
+
+Returns a Presigned URL to upload directly to S3.  
+The URL is only valid for a specified period of time. (Configurable in the [kuzzlerc file](https://docs.kuzzle.io/plugins/1/manual-setup/config/))
+
+File uploaded to the generated URL must be validated with `validateUpload` otherwise they will be deleted after the same TTL as for the URL expiration.
+
+*Request format:*
+
+```javascript
+{
+  // Kuzzle API params
+  "controller": "kuzzle-plugin-s3-upload/file",
+  "action": "getUploadUrl",
+
+  // Uploaded file name
+  "filename": "headcrab.png", 
+  // Upload directory
+  "uploadDir": "xen" 
+}
 ```
 
-#### Working on a different Kuzzle tag
+*Response result format:* 
 
-You can choose to work on the Kuzzle development branch by defining the following environment variables before launching `docker-compose`:
+```javascript
+{
+  // File key in S3 bucket
+  "fileKey": "xen/<uuid>/headcrab.png", 
+  // Presigned upload URL
+  "uploadUrl": "https://s3.eu-west-3.amazonaws.com/...", 
+  // Public file URL after successful upload
+  "fileUrl": "https://s3.eu-west-3.amazonaws.com/...", 
+  // TTL in ms for the URL validity and before the uploaded file deletion
+  "ttl": 600000 
+}
+```
 
-```bash
-$ export KUZZLE_DOCKER_TAG=1.4.2
-$ docker-compose -f docker/docker-compose.yml up
+#### *file:validateUpload*
+
+Validate and persist a previsously uploaded file.  
+Without a call to the action, every file uploaded on a Presigned URL will be deleted after a TTL.
+
+*Request format:*
+
+```javascript
+{
+  // Kuzzle API params
+  "controller": "kuzzle-plugin-s3-upload/file",
+  "action": "validateUpload",
+
+  // File key in S3 bucket
+  "fileKey": "xen/<uuid>/headcrab.png" 
+}
 ```
 
-These environment variables enable you to specify any existing build tag available on [Docker Hub](https://hub.docker.com/r/kuzzleio/kuzzle/tags/).
+*Response result format:* 
 
+```javascript
+{
+  // File key in S3 bucket
+  "fileKey": "xen/<uuid>/headcrab.png", 
+  // Public file URL after successful upload
+  "fileUrl": "https://s3.eu-west-3.amazonaws.com/..." 
+}
+```
 
-#### Customizing the plugin instance name
+#### *file:getUrl*
 
-You may want to choose a plugin name rather than using the default one. To do so, edit the `manifest.json` file at the root of this repo and change the `name` property to your convenience.
+Returns the public file URL.
 
+*Request format:*
 
-## `manifest.json` file
+```javascript
+{
+  // Kuzzle API params
+  "controller": "kuzzle-plugin-s3-upload/file",
+  "action": "getUrl",
 
-The `manifest.json` file is here to provide a description of your plugin to Kuzzle:
+  // File key in S3 bucket
+  "fileKey": "xen/<uuid>/headcrab.png" 
+}
+```
+
+*Response result format:* 
+
+```javascript
+{
+  // Public file URL after successful upload
+  "fileUrl": "https://s3.eu-west-3.amazonaws.com/..." 
+}
+```
+
+#### *file:deleteFile*
+
+Deletes an uploaded file from S3.
```

Comment:
... but isn't that plugin only about uploads? :trollface: 

More seriously though, this pleads in favor of renaming the plugin "s3" instead of "s3-upload"